File: backendmealplan/src/main/java/org/backendmealplan/backendmealplan/other/MealDTO.java
package org.backendmealplan.backendmealplan.other;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class MealDTO {

    private String mealName;
    private String imageUrl;
    private int calories;
    private String instructions;
    private String prepareTime;
    private String cookTime;
    private double fat;
    private double protein;
    private double carbs;
    private double fibre;
    private String tips;

    List<String> dietTypes = new ArrayList<>();
    List<IngredientDTO> ingredients = new ArrayList<>();


    List<DayMealDTO> dayMealDTOList = new ArrayList<>();

}

File: frontendmealplan/src/app/mealplan/models/MealDTO.ts
import { DayMealDTO } from "./DayMealDTO";
import { IngredientDTO } from "./IngredientDTO";

export class MealDTO {
    constructor(
        public mealName?: string,
        public imageUrl?: string,
        public calories?: number,
        public instructions?: string,
        public prepareTime?: string,
        public cookTime?: string,
        public fat?: number,
        public protein?: number,
        public carbs?: number,
        public fibre?: number,
        public tips?: string,
        public dietTypes?: string[],
        public ingredients?: IngredientDTO[],
        public dayMealDTOList?: DayMealDTO[]
    ) {}
}



